Scripts/RemediationScripts/Remediate-EnableEncryptionInTransitForStorageAccounts.ps1

```diff
@@ -212,16 +254,40 @@ function Enable-StorageEncryptionInTransit
             break
         }
     }
-  
-    #Connect to AzAccount
-    $isContextSet = Get-AzContext
 
-    if ([string]::IsNullOrEmpty($isContextSet))
-    {  
-        Write-Host $([Constants]::SingleDashLine)    
-        Write-Host "Connecting to AzAccount..."
-        Connect-AzAccount -ErrorAction Stop
-        Write-Host "Connected to AzAccount" -ForegroundColor $([Constants]::MessageType.Update)
+    $isContextSet = Get-AzContext
+    if ($UseSystemAssignedManagedIdentity)
+    {
+        # Connecting to AzAccount using system assigned managed identity, if, 1. Context is not set to any account. or 2. Context is already set with 'User' account type.
+        if (([string]::IsNullOrEmpty($isContextSet)) -or (![string]::IsNullOrEmpty($isContextSet) -and $isContextSet.Account.Type -eq "User"))
```

Comment:
No need to check for empty again, ![string]::IsNullOrEmpty($isContextSet)

Reply:
Makes sense.
Updated the condition.